### ISSUE TITLE
Laggy on games 100k plus

### ISSUE THREAD

**Issue by Mtihc** [Mtihc/bustabit-script-simulator#1]
The page becomes laggy when you simulate 100k+ games. Needs a performance fix.

**Comment by Mtihc** [comment]
Javscript does not support multithreading.

The obvious solution is to use WebWorkers. But since this project was bootstrapped with [create-react-app](https://github.com/facebook/create-react-app), this is not so easy. I'll have to do `npm run eject` or use [react-app-rewired](https://github.com/timarney/react-app-rewired) or something, in order to configure [worker-loader](https://www.npmjs.com/package/worker-loader).

**Comment by b3n3tt3** [comment]
i have a idea

pre calculate the 1.4mil games (using provably fair) and store the busts somewhere. Each hash's busts is readily available for anyone. Saves time if done this way instead of calculating from hash on every session

**Comment by Mtihc** [comment]
Calculating the busts is not really the heavy part. Simulating the games one by one is the heavy part!

I've fixed this using javascript's `setImmediate`. Now the browser won't freeze anymore.

**Comment by Mtihc** [comment]
Reopening this issue because it's still laggy at 100k+ simulated games.

Hoping this pull-request is merged soon: https://github.com/facebook/create-react-app/pull/5886 so that I can use WebWorkers to solve this issue.

**Comment by nikereds** [comment]
I have a personal question. I would like to ask for help. If possible, please reply to my email.

praise82t@gmail.com

**Comment by arigatou** [comment]
@Mtihc 

https://kentcdodds.com/blog/speed-up-your-app-with-web-workers

**Comment by izikazem** [comment]
Why not get a hash list with python and then send it to javascript? In python we can get 1 million numbers in seconds and js should just run the simulation.

**Comment by Mtihc** [comment]
You’re right, but...

This project is frontend only. If I want to have some socket messaging
between python and javascript I would need a backend and I can no longer
host it on github.io

Thats one argument against your suggestion.

**Comment by Mtihc** [comment]
Send me some cash and I might be able to afford a server ;)

**Comment by izikazem** [comment]
That is right, how about to add an option that allow to use a json file with for example, a list exported with python by the user. That's like if the user execute the "backend".
 I can develop the python's code as support.

**Comment by izhzz** [comment]
> I can develop the python's code as support.

All we need is a pre-made file with all the hashes for millions of games. We can just verifiy the pre-made file much quicker than collecting it by ourselves every time we need it, correct?

If you can supply a solutions like that, that would be great. Would sponsor your some coffees.

**Comment by Mtihc** [comment]
Seems like we’re going in circles.

Generating the hashes and the busts is not the heavy part. Actually
simulation the games with your custom script is the heavy part! So a
pre-made list of hashes will not solve the problem.

izikazem suggested simulating the games in python. And then I could make a
feature to import the results and render the chart. But that seems dumb.
You could just as easy use excel or something to render a chart.

**Comment by Mtihc** [comment]
In my opinion the only real solution is what I said here: https://github.com/Mtihc/bustabit-script-simulator/issues/1#issuecomment-480859644
